apps/meteor/server/startup/migrations/v305.ts

```diff
@@ -0,0 +1,57 @@
+import { LivechatInquiryStatus } from '@rocket.chat/core-typings';
+import { LivechatInquiry, LivechatRooms, Subscriptions } from '@rocket.chat/models';
+import type { UpdateResult } from 'mongodb';
+
+import { addMigration } from '../../lib/migrations';
+
+addMigration({
+	version: 305,
+	name: 'Remove Omnichannel stuck rooms caused by improper room closing or queue processing',
+	async up() {
+		// This migration aims to remove all the rooms that were left in an imporper state due to the closing process
+		// not being completed or the queue not updating them properly.
+		// Process is as follows:
+		// 1. Find all the closed rooms that still have an open inquiry
+		// 2. Remove the inquiry
+		// 1. Find all the inquiries that point to an invalid rooms
+		// 2. Remove the inquiry
+		// 1. Find all the inquiries that are queued but have any "takenAt" date
+		// 2. Remove the inquiry
+		// 1. Find all closed rooms
+		// 2. Remove subscriptions associated with them
+
+		// Closed rooms with open inquiries
+		const closedRooms = await LivechatRooms.find({ closedAt: { $exists: true } }, { projection: { _id: 1 } }).toArray();
+		const { deletedCount } = await LivechatInquiry.deleteMany({ rid: { $in: closedRooms.map((room) => room._id) } });
```

Comment:
With same case above this could fail because of the size of the array being returned